tools/c7n_org/c7n_org/cli.py

```diff
@@ -385,24 +388,30 @@ def report(config, output, use, output_dir, accounts,
     writer.writerows(rows)
 
 
-def _get_env_creds(session, region):
-    creds = session._session.get_credentials()
+def _get_env_creds(account, session, region):
     env = {}
-    env['AWS_ACCESS_KEY_ID'] = creds.access_key
-    env['AWS_SECRET_ACCESS_KEY'] = creds.secret_key
-    env['AWS_SESSION_TOKEN'] = creds.token
-    env['AWS_DEFAULT_REGION'] = region
+    if account["provider"] == 'aws':
+        creds = session._session.get_credentials()
+        env['AWS_ACCESS_KEY_ID'] = creds.access_key
+        env['AWS_SECRET_ACCESS_KEY'] = creds.secret_key
+        env['AWS_SESSION_TOKEN'] = creds.token
+        env['AWS_DEFAULT_REGION'] = region
+    elif account["provider"] == 'azure':
+        env['AZURE_SUBSCRIPTION_ID'] = account["account_id"]
+    elif account["provider"] == 'gcp':
+        env['PROJECT_ID'] = account["account_id"]
```

Comment:
gcp is a little more dysfunctional than that, there's like five env vars that are valid and different tools use different ones.. afaics we should be fine if we just set 'GOOGLE_CLOUD_PROJECT', 'CLOUDSDK_CORE_PROJECT'

Reply:
@kapilt never(really) used GCP, maybe we can set both "PROJECT_ID" and "GOOGLE_CLOUD_PROJECT"?

Reply:
PROJECT_ID isn't something that gcp sdk's use though, we should set two that will cover off on sdk and cli usage, ie both 'GOOGLE_CLOUD_PROJECT', 'CLOUDSDK_CORE_PROJECT'

Reply:
Ok updated